products/pluto/pluto_build/sql/zoning_zoningdistrict.sql

```diff
@@ -1,147 +1,188 @@
+-- calculate how much (total area and percentage) of each lot is covered by a zoning district
+-- assign the zoning district(s) to each tax lot
+-- the order zoning districts are assigned is based on which district covers the majority of the lot
+-- a district is only assigned if more than 10% of the district covers the lot
+-- OR the majority of the district is within the lot
+
+
 DROP TABLE IF EXISTS validdtm;
 CREATE TABLE validdtm AS (
-SELECT a.bbl, ST_MakeValid(a.geom) as geom 
-FROM pluto a
-WHERE ST_GeometryType(ST_MakeValid(a.geom)) = 'ST_MultiPolygon');
+SELECT bbl, ST_MakeValid(geom) AS geom 
+FROM pluto
+WHERE ST_GeometryType(ST_MakeValid(geom)) = 'ST_MultiPolygon');
 CREATE INDEX validdtm_geom_idx ON validdtm USING GIST (geom gist_geometry_ops_2d);
 
 ANALYZE validdtm;
 
 DROP TABLE IF EXISTS validzones;
 CREATE TABLE validzones AS (
-SELECT a.zonedist, ST_MakeValid(a.geom) as geom  
-FROM dcp_zoningdistricts a
-WHERE ST_GeometryType(ST_MakeValid(a.geom)) = 'ST_MultiPolygon'); 
+SELECT 
+    CASE
+        WHEN zonedist = ANY('{"BALL FIELD", "PLAYGROUND", "PUBLIC SPACE"}') THEN 'PARK'
+        ELSE zonedist
+    END AS zonedist, 
+    ST_MakeValid(geom) AS geom  
+FROM dcp_zoningdistricts
+WHERE ST_GeometryType(ST_MakeValid(geom)) = 'ST_MultiPolygon'); 
+
 CREATE INDEX validzones_geom_idx ON validzones USING GIST (geom gist_geometry_ops_2d);
 
 ANALYZE validzones;
 
 DROP TABLE IF EXISTS lotzoneper;
-CREATE TABLE lotzoneper AS (
-    SELECT p.bbl, n.zonedist
-    , (ST_Area(
+CREATE TABLE lotzoneper AS 
+SELECT 
+    p.bbl, 
+    n.zonedist, 
+    ST_Area(
         CASE 
             WHEN ST_CoveredBy(p.geom, n.geom) THEN p.geom::geography
             ELSE ST_Multi(ST_Intersection(p.geom,n.geom))::geography
```

Comment:
@fvankrieken (cc @damonmcc ) potentially silly question, but is ST_Intersection not commutative? Wondering why we have both `segbblgeom` and `segzonegeom`

Reply:
That is a great point, it definitely is